docker-compose.yml

```diff
@@ -507,12 +583,70 @@ services:
       retries: 5
       timeout: 5s
     depends_on:
+      router:
+        condition: "service_started"
       api:
         condition: "service_healthy"
+    networks:
+      relays:
+        ipv4_address: ${RELAY_2_PUBLIC_IP4_ADDR:-172.29.0.102}
+        ipv6_address: ${RELAY_2_PUBLIC_IP6_ADDR:-172:29:0::102}
+
+  # Relays in prod always talk to a router to reach the Internet. We leverage this to avoid a map lookup and simply swap the
+  # MACs for all relayed traffic. So we mimic this setup for local dev and CI to ensure this eBPF code path is getting exercised.
+  # For this to work, we need to ensure the relays and client/gateway are *not* connected to the same Docker network, otherwise
+  # they will learn each other's MAC addresses via ARP and the next-hop MAC swap will not be valid.
+  router:
+    image: alpine:3.22
+    sysctls:
+      - net.ipv4.ip_forward=1
+      - net.ipv6.conf.all.forwarding=1
+      - net.ipv6.conf.default.forwarding=1
+      - net.ipv6.conf.all.disable_ipv6=0
+      - net.ipv6.conf.default.disable_ipv6=0
+    command: ["sleep", "infinity"]
+    init: true
     networks:
       app:
-        ipv4_address: ${RELAY_2_PUBLIC_IP4_ADDR:-172.28.0.201}
-        ipv6_address: ${RELAY_2_PUBLIC_IP6_ADDR:-172:28:0::201}
+        ipv4_address: 172.28.0.254
+        ipv6_address: 172:28:0::254
+      relays:
+        ipv4_address: 172.29.0.254
+        ipv6_address: 172:29:0::254
+
+  # The veth driver uses a pair of interfaces to connect the docker bridge to the container namespace.
```

Comment:
This was quite the rabbit hole, and took lots of digging to uncover. Once you peel back the layers in play, it makes sense though.

Once I realized the XDP_TX was dropping before "hitting" the wire (showing in TX drops on eth0), it became clear I needed to either enable the GRO workaround or add a dummy XDP_PASS program.

After several hours wrestling with heavily delayed packet bursts using the GRO implementation, I turned to just building a Docker image we can use in CI that simply does XDP_PASS, and we attach this to the host-side veth pairs so that the Docker bridge sees the packets sent by the Relays, and all is fine.

Related: https://github.com/xdp-project/xdp-tutorial/issues/399